client/src/components/Verejne/index.js

```diff
@@ -45,20 +70,25 @@ export default compose(
     (state) => ({
       entities: entitiesSelector(state),
     }),
-    {fetchEntities},
+    {fetchEntities, selectEntity}
   ),
   lifecycle({
     componentDidMount() {
       this.props.fetchEntities()
     },
   }),
   withState('currentPage', 'setCurrentPage', 1),
-  withProps(({entities, currentPage}) => ({
-    entities: chunk(reverse(sortBy(entities, ['size'])), VEREJNE_MAX_PAGE_ITEMS)[currentPage - 1] || [],
-    pageCount: entities ? Math.ceil(entities.length / VEREJNE_MAX_PAGE_ITEMS) : 0,
-  })),
-  branch(
-    (props) => !props.entities.length,
-    renderComponent(Loading),
-  )
+  withProps(({entities, currentPage}) => {
+    let newEntities = entities
+    if (entities && entities.length) {
+      newEntities = chunk(reverse(sortBy(entities, ['size'])), VEREJNE_MAX_PAGE_ITEMS)[
+        currentPage - 1
+      ]
+    }
+    return {
+      entities: newEntities,
+      pageCount: entities ? Math.ceil(entities.length / VEREJNE_MAX_PAGE_ITEMS) : 0,
+    }
+  }),
+  branch((props) => !props.entities, renderComponent(Loading))
```

Comment:
this is nice 👍 .. though it'll get handled by data-providers once they're here, good to know